ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfiguration.java

```diff
@@ -0,0 +1,121 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
+
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.INTERNAL_ERROR;
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.INVALID_PARAMS;
+import static org.hyperledger.besu.util.Slf4jLambdaHelper.traceLambda;
+
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.EngineExchangeTransitionConfigurationParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.EngineExchangeTransitionConfigurationResult;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+
+import java.util.Optional;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EngineExchangeTransitionConfiguration extends ExecutionEngineJsonRpcMethod {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(EngineExchangeTransitionConfiguration.class);
+
+  public EngineExchangeTransitionConfiguration(
+      final Vertx vertx, final ProtocolContext protocolContext) {
+    super(vertx, protocolContext);
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ENGINE_EXCHANGE_TRANSITION_CONFIGURATION.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final EngineExchangeTransitionConfigurationParameter remoteTransitionConfiguration =
+        requestContext.getRequiredParameter(
+            0, EngineExchangeTransitionConfigurationParameter.class);
+    final Object reqId = requestContext.getRequest().getId();
+
+    traceLambda(
+        LOG,
+        "received transitionConfiguration: {}",
+        () -> Json.encodePrettily(remoteTransitionConfiguration));
+
+    if (remoteTransitionConfiguration.getTerminalBlockNumber() != 0) {
+      return respondWithError(reqId, INVALID_PARAMS);
+    }
+
+    final Optional<BlockHeader> maybeFinalizedBlockHeader = mergeContext.getFinalized();
```

Comment:
Should not this be `getTerminalPoWBlock` instead of `getFinalized`?

Reply:
Yes, I think you are right. I've reviewed everything again and it needs to be indeed getTerminalPoWBlock